src/orion/core/utils/tests.py

```diff
@@ -146,45 +178,97 @@ def get_trial(self, index):
 
     def cleanup(self):
         """Cleanup after testing"""
-        self.database.remove('experiments', {})
-        self.database.remove('trials', {})
+        if self.database:
```

Comment:
Should we create child classes of OrionState for Legacy and Track instead? Handling both in the same class seems to be leading to some hacky fixes.